src/System.Private.CoreLib/shared/System/Text/UnicodeEncoding.cs

```diff
@@ -452,7 +455,8 @@ internal override unsafe int GetByteCount(char* chars, int count, EncoderNLS enc
 
                                     // If they happen to be high/low/high/low, we may as well continue.  Check the next
                                     // bit to see if its set (low) or not (high) in the right pattern
-                                    if (((0xfc00fc00fc00fc00 & *longChars) ^ highLowPatternMask) != 0)
+                                    if ((0xfc00fc00fc00fc00 & *longChars) !=
+                                            (BitConverter.IsLittleEndian ? (ulong)0xdc00d800dc00d800 : (ulong)0xd800dc00d800dc00))
```

Comment:
>(BitConverter.IsLittleEndian ? (ulong)0xdc00d800dc00d800 : (ulong)0xd800dc00d800dc00) [](start = 44, length = 85)

can we just cache this value and avoid the checks when it gets used?

Reply:
The IsLittleEndian check is optimized out by the JIT.

The current code that is caching it in readonly variable has worse performance. This caching was introduced just recently in #18254. This change is fixing the performance regression introduced by #18254.